### ISSUE TITLE
Support for current hash algorithms

### ISSUE THREAD

**Issue by chcg** [bruderstein/npppm3#14]
See https://en.wikipedia.org/wiki/MD5#Security
and https://en.wikipedia.org/wiki/SHA-1#Attacks, so maybe support sha2 or sha3.

On client side probalby just sha1 is available, see
https://msdn.microsoft.com/de-de/library/windows/desktop/bb931357(v=vs.85).aspx
I will have a deeper look at that.

**Comment by bruderstein** [comment]
Yes, i am aware of the security implications of MD5. But we'd need to support more on the server side (the server that validates) _and_ in the client (PM).  Once we have this MVP, then we can add to the functionality, and select some better hashing algorithms.

The attack vectors are very limited here though, it's more used just to identify that the correct file has been downloaded, and we can quickly mark a plugin file as bad if it starts doing evil things.

**Comment by chcg** [comment]
Yes, that is a future task after MVP. Keep just in mind as N++ is part of wikileaks https://notepad-plus-plus.org/community/topic/13402/dll-hack-in-notepad/

**Comment by chcg** [comment]
Unforunately to run the docker stuff on windows 
https://store.docker.com/editions/community/docker-ce-desktop-windows?tab=description

the win10 professional edition is needed. Just have the home variant and what is even worse my cpu doesn't support vitualization, so also the https://github.com/docker/toolbox is no option or a Linux VM :-(

**Comment by bruderstein** [comment]
Windows issues moved to #17
